common/Cargo.toml

```diff
@@ -1,12 +1,29 @@
 [package]
```

Comment:
It seems like the major goal of this PR is moving much of the core consensus verification functions into it's own crate? Right now that crate is common.

I prefer if we keep common as lightweight as possible and be things that are used by many crates and must be placed in common to avoid the circular import rule. I think the better solution would be to move the core consensus primitives into some new crate `consensus-core` which provers can then consume.